src/Morphir/IR/Repo.elm

```diff
@@ -131,14 +168,205 @@ orderElmModulesByDependency parsedModules =
+                                                                    (\argType ->
+                                                                        ( [ "arg", String.fromInt (index + 1) ]
+                                                                        , argType
+                                                                        )
+                                                                    )
+                                                        )
+                                                    |> ResultList.keepAllErrors
+                                                    |> Result.mapError List.concat
+                                        in
+                                        constructorArgsResult
+                                            |> Result.map
+                                                (\constructorArgs ->
+                                                    ( constructorName, constructorArgs )
+                                                )
+                                    )
+                                |> ResultList.keepAllErrors
+                                |> Result.map Dict.fromList
+                                |> Result.mapError List.concat
+
+                        withAccessControl : Bool -> a -> AccessControlled a
+                        withAccessControl isExposed value =
+                            if isExposed then
+                                AccessControlled.public value
+
+                            else
+                                AccessControlled.private value
+                    in
+                    constructorsResult
+                        |> Result.map
+                            (\constructors ->
+                                ( customType.name |> Node.value |> Name.fromString
+                                , Type.customTypeDefinition typeParams (withAccessControl True constructors)
+                                )
+                            )
+                        |> Just
+
+                AliasDeclaration typeAlias ->
+                    let
+                        typeParams : List Name
+                        typeParams =
+                            typeAlias.generics
+                                |> List.map (Node.value >> Name.fromString)
+                    in
+                    typeAlias.typeAnnotation
+                        |> mapAnnotationToMorphirType
+                        |> Result.map
+                            (\tpe ->
+                                ( typeAlias.name
+                                    |> Node.value
+                                    |> Name.fromString
+                                , Type.TypeAliasDefinition typeParams tpe
+                                )
+                            )
+                        |> Just
+
+                _ ->
+                    Nothing
+    in
     Debug.todo "implement"
```

Comment:
Don't you want to return `typeNameToDefinition` here?

Reply:
Done. Don't know how I managed to miss that.